eslint 2.10.1 - Parsing error: Unexpected token

**What version of ESLint are you using?**
2.10.1

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**
```
module.exports = {
    "extends": "airbnb",
    "parser": "babel-eslint",
}
```

**What did you do? Please include the actual source code causing the issue.**

I have a sample demonstrating the issue at:
https://github.com/michael-wolfenden/eslint-2-10-1-issue

but running eslint via webpack with:

```
export class Counter {
  static initialCount = 0;
}
```

generates the error:

```
C:\Dev\eslint-issue\Counter.js
  2:23  error  Parsing error: Unexpected token =
```

**What did you expect to happen?**
No errors

**What actually happened? Please include the actual, raw output from ESLint.**
```
C:\Dev\eslint-issue\Counter.js
  2:23  error  Parsing error: Unexpected token =
```

## Addition information

The issue first occurred with `2.10.0` so at first I assumed it was the same issue as #6158 , however upgrading to `2.10.1` didn't fix the issue. It appears as though the eslint babel parser isn't kicking in.

Another thing to note is that the issue only occurs when running eslint via webpack.

I'm getting a different issue in the demo repo, which is making it a bit harder to debug:

```
$ npm start

> eslint-issue@1.0.0 start C:\Users\Nicholas\projects\personal\eslint-2-10-1-issue
> webpack

Hash: 396f0bfb9d565b6f60f0
Version: webpack 1.13.0
Time: 827ms
    + 1 hidden modules

ERROR in ./Counter.js
Module build failed: Error: Failed to load plugin jsx-a11y: Cannot find module 'eslint-plugin-jsx-a11y'
Referenced from: airbnb
Referenced from: C:\Users\Nicholas\projects\personal\eslint-2-10-1-issue\.eslintrc.js
    at Function.Module._resolveFilename (module.js:325:15)
    at Function.Module._load (module.js:276:25)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.module.exports.load (C:\Users\Nicholas\projects\personal\eslint\lib\config\plugins.js:114:26)
    at Array.forEach (native)
    at Object.module.exports.loadAll (C:\Users\Nicholas\projects\personal\eslint\lib\config\plugins.js:136:21)
    at load (C:\Users\Nicholas\projects\personal\eslint\lib\config\config-file.js:500:21)
    at C:\Users\Nicholas\projects\personal\eslint\lib\config\config-file.js:386:36
    at Array.reduceRight (native)
    at applyExtends (C:\Users\Nicholas\projects\personal\eslint\lib\config\config-file.js:363:28)
    at load (C:\Users\Nicholas\projects\personal\eslint\lib\config\config-file.js:525:22)
    at C:\Users\Nicholas\projects\personal\eslint\lib\config\config-file.js:386:36
    at Array.reduceRight (native)
    at applyExtends (C:\Users\Nicholas\projects\personal\eslint\lib\config\config-file.js:363:28)
    at Object.load (C:\Users\Nicholas\projects\personal\eslint\lib\config\config-file.js:525:22)
    at loadConfig (C:\Users\Nicholas\projects\personal\eslint\lib\config.js:64:33)
    at getLocalConfig (C:\Users\Nicholas\projects\personal\eslint\lib\config.js:126:23)
    at Config.getConfig (C:\Users\Nicholas\projects\personal\eslint\lib\config.js:227:22)
    at processText (C:\Users\Nicholas\projects\personal\eslint\lib\cli-engine.js:217:27)
    at CLIEngine.executeOnText (C:\Users\Nicholas\projects\personal\eslint\lib\cli-engine.js:755:26)
    at lint (C:\Users\Nicholas\projects\personal\eslint-2-10-1-issue\node_modules\eslint-loader\index.js:25:20)
    at Object.module.exports (C:\Users\Nicholas\projects\personal\eslint-2-10-1-issue\node_modules\eslint-loader\index.js:111:3)
```

--- a/conf/cli-options.js
+++ b/conf/cli-options.js
@@ -16,6 +16,7 @@ module.exports = {
     extensions: [".js"],
     ignore: true,
     ignorePath: null,
+    parser: "",     // must be empty
     cache: false,
 
     // in order to honor the cacheFile option if specified

--- a/conf/cli-options.js
+++ b/conf/cli-options.js
@@ -5,8 +5,6 @@
 
 "use strict";
 
-var DEFAULT_PARSER = require("../conf/eslint.json").parser;
-
 module.exports = {
     configFile: null,
     baseConfig: false,
@@ -18,8 +16,8 @@ module.exports = {
     extensions: [".js"],
     ignore: true,
     ignorePath: null,
-    parser: DEFAULT_PARSER,
     cache: false,
+
     // in order to honor the cacheFile option if specified
     // this option should not have a default value otherwise
     // it will always be used

--- a/tests/fixtures/configurations/parser/.eslintrc.json
+++ b/tests/fixtures/configurations/parser/.eslintrc.json
@@ -1,0 +1,7 @@
+{
+    "root": true,
+    "parser": "./custom.js",
+    "rules": {
+        "quotes": ["error", "double"]
+    }
+}

--- a/tests/fixtures/configurations/parser/custom.js
+++ b/tests/fixtures/configurations/parser/custom.js
@@ -1,0 +1,3 @@
+exports.parse = function() {
+    throw new Error("Boom!");
+};

--- a/tests/lib/cli-engine.js
+++ b/tests/lib/cli-engine.js
@@ -274,6 +274,23 @@ describe("CLIEngine", function() {
 
         var engine;
 
+        it("should use correct parser when custom parser is specified", function() {
+
+            engine = new CLIEngine({
+                cwd: originalDir,
+                ignore: false
+            });
+
+            var filePath = path.resolve(__dirname, "../fixtures/configurations/parser/custom.js");
+            var report = engine.executeOnFiles([filePath]);
+
+            assert.equal(report.results.length, 1);
+            assert.equal(report.results[0].messages.length, 1);
+            assert.equal(report.results[0].messages[0].message, "Parsing error: Boom!");
+
+        });
+
+
         it("should report zero messages when given a config file and a valid file", function() {
 
             engine = new CLIEngine({

--- a/tests/lib/config.js
+++ b/tests/lib/config.js
@@ -2,7 +2,7 @@
  * @fileoverview Tests for config object.
  * @author Seth McLaughlin
  */
-
+/* eslint no-undefined: "off" */
 "use strict";
 
 //------------------------------------------------------------------------------
@@ -289,6 +289,13 @@ describe("Config", function() {
             assert.equal(noUndef, 2);
         });
 
+        it("should contain the correct value for parser when a custom parser is specified", function() {
+            var configPath = path.resolve(__dirname, "../fixtures/configurations/parser/.eslintrc.json"),
+                configHelper = new Config({ cwd: process.cwd() }),
+                config = configHelper.getConfig(configPath);
+
+            assert.equal(config.parser, path.resolve(path.dirname(configPath), "./custom.js"));
+        });
 
         // Configuration hierarchy ---------------------------------------------
 
@@ -752,7 +759,7 @@ describe("Config", function() {
                         parserOptions: {},
                         env: {},
                         globals: {},
-                        parser: void 0,
+                        parser: undefined,
                         rules: {
                             "home-folder-rule": 2
                         }
@@ -776,7 +783,7 @@ describe("Config", function() {
                         parserOptions: {},
                         env: {},
                         globals: {},
-                        parser: void 0,
+                        parser: undefined,
                         rules: {
                             "project-level-rule": 2
                         }
@@ -801,7 +808,7 @@ describe("Config", function() {
                         parserOptions: {},
                         env: {},
                         globals: {},
-                        parser: void 0,
+                        parser: undefined,
                         rules: {
                             quotes: [2, "double"]
                         }
@@ -825,7 +832,7 @@ describe("Config", function() {
                         parserOptions: {},
                         env: {},
                         globals: {},
-                        parser: void 0,
+                        parser: undefined,
                         rules: {}
                     };
 
@@ -847,7 +854,7 @@ describe("Config", function() {
                         parserOptions: {},
                         env: {},
                         globals: {},
-                        parser: void 0,
+                        parser: undefined,
                         rules: {}
                     };
 
@@ -868,7 +875,7 @@ describe("Config", function() {
                         parserOptions: {},
                         env: {},
                         globals: {},
-                        parser: void 0,
+                        parser: undefined,
                         rules: {
                             "project-level-rule": 2,
                             "subfolder-level-rule": 2

